### DISCUSSION TITLE
Switching clock source at runtime

### DISCUSSION THREAD

**Discussion by jefflongo** [zephyrproject-rtos/zephyr#55421] [Q&A]
I have a use-case with STM32 where I need to boot by driving the internal PLL using the HSI oscillator, then after configuring some peripherals, switch the PLL to use the HSE oscillator. Is this possible to do in Zephyr?

**Comment by erwango** [comment]
ATM this is not possible, sorry

**Comment by simontheu** [comment]
I did this with atmel ok. I set up my external clock with I2C then switched to it after.
